x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/search_experience.tsx

```diff
@@ -88,15 +87,15 @@ export const SearchExperience: React.FC = () => {
   const connector = new AppSearchAPIConnector({
     cacheResponses: false,
     endpointBase,
-    engineName: engine.name,
+    engineName: engine.name as string,
```

Comment:
I didn't fully figure out the error here. My guess is that `engine.name` from logic file _theoretically_ could be undefined. But since the search experience is only rendered inside the engine, I think we can safely ignore this case.